holidays/holiday_base.py

```diff
@@ -1041,7 +1041,9 @@ def pop_named(self, name: str) -> list[date]:
             KeyError if date is not a holiday and default is not given.
         """
         use_exact_name = HOLIDAY_NAME_DELIMITER in name
-        if not (dts := self.get_named(name, split_multiple_names=not use_exact_name)):
+        if not (
+            dts := self.get_named(name, lookup=lookup, split_multiple_names=not use_exact_name)
```

Comment:
It'd be also great to have some [tests](https://github.com/vacanza/holidays/blob/93b3ca994342dd79b509c23aaad28ed89dc8543c/tests/test_holiday_base.py#L778) added too